Rule parity with SQLFluff (#9778)
fishtown-analytics/docs.getdbt.com: website/docs/reference/commands/lint.md

Reviewer: ```suggestion
`dbt lint` is part of the <Constant name="fusion_engine" />. It is not the same as `dbt sqlfluff lint` on the <Constant name="platform_cli" />. For SQLFluff on the platform CLI, see [Configure the dbt CLI](/docs/platform/configure-dbt-cli). [Linting in Studio IDE](/docs/platform/studio-ide/lint-format) continues to use SQLFluff.
```

---
title: "About dbt lint command"
sidebar_label: "lint"

> description: "Use dbt lint to check your SQL files for style, correctness, and convention issues using your `.sqlfluff` config and SQLFluff rule codes."

Author: ```suggestion
description: "Use dbt lint to check your SQL files for style, correctness, and convention violations using your `.sqlfluff` config and SQLFluff rule codes."
```

> description: "Use dbt lint to check your SQL files for style, correctness, and convention violations using your `.sqlfluff` config and SQLFluff rule codes."

Reviewer: backticks don't render in frontmatter
```suggestion
description: "Use dbt lint to check your SQL files for style, correctness, and convention violations using your .sqlfluff config and SQLFluff rule codes."
```

id: "lint"
availability:
  engine: v2
---

# About dbt lint command <Lifecycle status="beta" />

> `dbt lint` is a high-performance SQL linter built into the dbt platform. It is SQLFluff-compatible: it reads your `.sqlfluff` config, uses the same rule codes (for example, `CP01`, `RF03`), and respects `-- noqa` suppression comments. Compatible does not mean identical. `dbt lint` and SQLFluff can still return different results for the same file and config.

Reviewer: ```suggestion
`dbt lint` is a high-performance SQL linter built into the dbt platform. It is SQLFluff-compatible: it reads your `.sqlfluff` config, uses the same rule codes (for example, `CP01`, `RF03`), and respects `-- noqa` suppression comments. Compatible does not mean identical: `dbt lint` and SQLFluff can return different results for the same file and config. Refer to [Rule parity with SQLFluff](#rule-parity-with-sqlfluff).
```

Keep these limitations in mind:

> ### Rule parity with SQLFluff

Reviewer: I suggest we move this to its own section and make it an H2. Putting it under "Beta limitations" suggests it will not be a limitation in GA, but reading Elias' response, that's not true.

> `dbt lint` aims for high overlap with SQLFluff, but it does not guarantee rule-for-rule parity. Small differences can still appear, including layout and indentation rules such as `LT02`. Because [linting in the <Constant name="studio_ide" />](/docs/platform/studio-ide/lint-format) still uses SQLFluff, <Constant name="studio_ide" /> **Lint** and `dbt lint` can disagree on the same project code. If you need exact SQLFluff behavior today, continue using SQLFluff in the <Constant name="studio_ide" /> or with the <Constant name="platform_cli" /> alongside <Constant name="fusion" />.

Reviewer: Small edits to make it clear this is not a temporary situation and to make it easier to read what applies to your situation.

```suggestion
`dbt lint` aims for high overlap with SQLFluff, but it doesn't guarantee rule-for-rule parity, and small differences will always exist. Layout and indentation rules, such as `LT02`, are one known area of difference.

Because [linting in the <Constant name="studio_ide" />](/docs/platform/studio-ide/lint-format) still uses SQLFluff, <Constant name="studio_ide" /> **Lint file** and `dbt lint` can report different violations for the same project code. Similarly, CI jobs on a <Constant name="fusion" /> version invoke `dbt lint` instead of SQLFluff, so results from [CI jobs](/docs/deploy/continuous-integration#sql-linting) can differ from your SQLFluff results.

If you need SQLFluff behavior, you can either lint in the <Constant name="studio_ide" />, which continues to run SQLFluff, or run SQLFluff locally using the standalone <Constant name="core" /> engine templater. Refer to [<Constant name="fusion" /> limitations](/docs/fusion/supported-features#limitations) for more information.
```

Reviewer: We should verify if this is oK and correct:
> If you need SQLFluff behavior, you can either lint in the <Constant name="studio_ide" />, which continues to run SQLFluff, or run SQLFluff locally using the standalone <Constant name="core" /> engine templater. Refer to [<Constant name="fusion" /> limitations](/docs/fusion/supported-features#limitations) for more information.
